core/src/hashes/blake2b256.rs

```diff
@@ -0,0 +1,92 @@
+use crate::hashes::marked::MarkedDigestWriter;
+use blake2_rfc::blake2b::Blake2b;
+use std::io::{Result as IOResult, Write};
+
+struct Blake2b256 {
+    ctx: Blake2b,
```

Comment:
the underlying `Blake2b` struct supports `Write` so I think we can cut out this middle-man struct. and simplify the `Write` implementation below

Reply:
Pushed up a new commit that cuts out the middle man struct, much cleaner now :+1: